MDANSE/Src/MDANSE/Framework/Jobs/TrajectoryFilter.py

```diff
@@ -0,0 +1,279 @@
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+import collections
+import copy
+import json
+
+import numpy as np
+
+import h5py
+from MDANSE.Framework.Formats import HDFFormat
+from MDANSE.Chemistry.ChemicalSystem import ChemicalSystem
+from MDANSE.Framework.Jobs.IJob import IJob
+from MDANSE.Mathematics.Signal import FILTER_MAP
+from MDANSE.MolecularDynamics.Configuration import (
+    RealConfiguration,
+    PeriodicRealConfiguration,
+)
+from MDANSE.MolecularDynamics.Trajectory import TrajectoryWriter
+from MDANSE.MLogging import LOG
+
+
+class TrajectoryFilter(IJob):
+    """
+    Design and apply a filter for the atomic trajectories.
+    """
+
+    label = "Trajectory Filter"
+
+    category = (
+        "Analysis",
+        "Trajectory",
+    )
+
+    ancestor = ["hdf_trajectory", "molecular_viewer"]
+
+    settings = collections.OrderedDict()
+    settings["trajectory"] = ("HDFTrajectoryConfigurator", {})
+    settings["frames"] = (
+        "CorrelationFramesConfigurator",
+        {"dependencies": {"trajectory": "trajectory"}},
+    )
+    settings["instrument_resolution"] = (
+        "InstrumentResolutionConfigurator",
+        {"dependencies": {"trajectory": "trajectory", "frames": "frames"}},
+    )
+    settings["projection"] = (
+        "ProjectionConfigurator",
+        {"label": "project coordinates"},
+    )
+    settings["trajectory_filter"] = (
+        "TrajectoryFilterConfigurator",
+        {"dependencies": {"trajectory": "trajectory"}},
+    )
+    settings["atom_selection"] = (
+        "AtomSelectionConfigurator",
+        {"dependencies": {"trajectory": "trajectory"}},
+    )
+    settings["weights"] = (
+        "WeightsConfigurator",
+        {
+            "default": "atomic_weight",
+            "dependencies": {
+                "trajectory": "trajectory",
+                "atom_selection": "atom_selection",
+            },
+        },
+    )
+    settings["output_files"] = (
+        "OutputTrajectoryConfigurator",
+        {"format": "MDTFormat"},
+    )
+    settings["running_mode"] = ("RunningModeConfigurator", {})
+
+    def initialize(self):
+        """
+        Initialize the input parameters and analysis self variables
+        """
+        super().initialize()
+
+        self.numberOfSteps = self.configuration["atom_selection"]["selection_length"]
+
+        self._atoms = self.configuration["trajectory"][
+            "instance"
+        ].chemical_system.atom_list
+
+        self._selected_atoms = [
+            self._atoms[i]
+            for i in np.array(self.configuration["atom_selection"]["indices"]).flatten()
```

Comment:
```suggestion
            for i in np.array(self.configuration["atom_selection"]["indices"]).flat
```
No need to copy the array again, already a lot of shuffling